v2/ical.NET/Serialization/SimpleDeserializer.cs

```diff
@@ -0,0 +1,221 @@
+            }
+        }
+
+        private CalendarProperty ParseContentLine(SerializationContext context, string input)
+        {
+            var match = _contentLineRegex.Match(input);
+            if (!match.Success)
+            {
+                throw new SerializationException($"Could not parse line: '{input}'");
+            }
+            var name = match.Groups[_nameGroup].Value;
+            var value = match.Groups[_valueGroup].Value;
+            var paramNames = match.Groups[_paramNameGroup].Captures;
+            var paramValues = match.Groups[_paramValueGroup].Captures;
+
+            var property = new CalendarProperty(name.ToUpperInvariant());
+            context.Push(property);
+            SetPropertyParameters(property, paramNames, paramValues);
+            SetPropertyValue(context, property, value);
+            context.Pop();
+            return property;
+        }
+
+        private static void SetPropertyParameters(CalendarProperty property, CaptureCollection paramNames, CaptureCollection paramValues)
+        {
+            var paramValueIndex = 0;
+            for (var paramNameIndex = 0; paramNameIndex < paramNames.Count; paramNameIndex++)
+            {
+                var paramName = paramNames[paramNameIndex].Value;
+                var parameter = new CalendarParameter(paramName);
+                var nextParamIndex = paramNameIndex + 1 < paramNames.Count ? paramNames[paramNameIndex + 1].Index : int.MaxValue;
+                while (paramValueIndex < paramValues.Count && paramValues[paramValueIndex].Index < nextParamIndex)
+                {
+                    var paramValue = paramValues[paramValueIndex].Value;
+                    parameter.AddValue(paramValue);
+                    paramValueIndex++;
+                }
+                property.AddParameter(parameter);
+            }
+        }
+
+        private void SetPropertyValue(SerializationContext context, CalendarProperty property, string value)
+        {
+            var type = _dataTypeMapper.GetPropertyMapping(property) ?? typeof(string);
+            var serializer = (SerializerBase)_serializerFactory.Build(type, context);
+            using (var valueReader = new StringReader(value))
+            {
+                var propertyValue = serializer.Deserialize(valueReader);
+                var propertyValues = propertyValue as IEnumerable<string>;
+                if (propertyValues != null)
+                {
+                    foreach (var singlePropertyValue in propertyValues)
+                    {
+                        property.AddValue(singlePropertyValue);
+                    }
+                }
+                else
+                {
+                    property.AddValue(propertyValue);
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetContentLines(TextReader reader)
+        {
+            var currentLine = new StringBuilder();
```

Comment:
I know we've disagreed on this in the past, but since this PR is about optimization... default capacity of `StringBuilder` is 16. The spec allows for line widths of 75 octets. Attachments are often many lines long, and normal calendar components are often more than 16 characters long as well, with, e.g. "BEGIN:" being 6 characters right off the bat. In order to avoid the realloc -> GC overhead that'll be incurred the majority of the time (16, 32, 64, 128 -- I tested), might be better to initialize the StringBuilder with a capacity of 75?